python/smqtk/detect_and_describe/__init__.py

```diff
@@ -0,0 +1,64 @@
+import abc
+import os
+
+from smqtk.data_rep import DataElement
+from smqtk.utils.plugin import get_plugins
+
+class DetectAndDescribe (object):
+	def __init__(self, descriptor_element_factory, data_element_type):
```

Comment:
I agree with your first point.

As for ``data_element_type``, that is there because it was the way I was using to generate the hash that is used as the UUID in creating the ``descriptor_element_factory``. Probably would make sense to pull this hashing out to a more general utility function.